### DISCUSSION TITLE
Not able to generate the trust chain of certificates with intermediate certificate

### DISCUSSION THREAD

**Discussion by BalajiVM** [rabbitmq/tls-gen#46] [General]
Hi ,

I have to generate trust chain of certificate with rootCA, intermediate CA and leaf certificates. 
With the two_shared_intermediates and separate_intermediates code, i am able to generate root CA and leaf certificate, where i am not able to find the intermediate certificate to form trust chain of certificate.
root certificate (ok) 
intermediate certificate (not available)
leaf certificate generated by intermediate certificate (ok)

Any help on this is really appreciable.

**Comment by lukebakken** [comment]
Please provide a script or series of commands that demonstrates what you are trying. I can't tell from your description. Thank you.

**Comment by BalajiVM** [comment]
cd tls_gen/
 cd separate_intermediates
 make USE_ECC=true ECC_CURVE="prime256v1"
 ls -lha ./result
 openssl ecparam -list_curves
 make verify
  make info

**Comment by BalajiVM** [comment]
I got this as an output 
![image](https://user-images.githubusercontent.com/12237365/173630598-58e7ed1f-14b3-45e8-b081-55df1eaa36e2.png)

**Comment by BalajiVM** [comment]
Inside the result folder i got the server and client certificates 

![image](https://user-images.githubusercontent.com/12237365/173630728-fc596905-f17b-49c7-86ba-c2979e071c69.png)

from testca i got the root CA certificate

![image](https://user-images.githubusercontent.com/12237365/173630984-33a805b7-8125-463a-8ebf-02aa053a3972.png)


But i couldn't find the intermediate certificate

**Comment by BalajiVM** [comment]
On the leaf certificate, the issuer is intermediate CA certificate. whereas, i didn't see any reference in the document about how to generate and install the intermediate certificate, and create trust chain of certificate.

**Reply by michaelklishin** [reply]
Concatenate them in order into a single file. Such files are called certificate bundles.

**Reply by BalajiVM** [reply]
any code reference is really appreciable :)

**Reply by michaelklishin** [reply]
> any code reference is really appreciable :)

Just google it https://serverfault.com/questions/476576/how-to-combine-various-certificates-into-single-pem
